structure/rbtree/rbtree.go

```diff
@@ -0,0 +1,457 @@
+// Red-Black Tree is a kind of self-balancing binary search tree.
+// Each node stores "color" ("red" or "black"), used to ensure that the tree remains balanced during insertions and deletions.
+//
+// For more details check out those links below here:
+// Programiz article : https://www.programiz.com/dsa/red-black-tree
+// Wikipedia article: https://en.wikipedia.org/wiki/Red_black_tree
+// authors [guuzaa](https://github.com/guuzaa)
+// see rbtree.go
+
+package rbtree
```

Comment:
I recommend moving both the binarysearchtree and rbtree package to be moved to a parent package called `tree` inside the `structure` package.

Reply:
Thanks for your suggestions. I will move `binarysearchtree` and `rbtree` package to `tree` package.

Reply:
If you can, yes do it. Though if I remember correctly there were some issues with `avl`, so you might come across some issues there. Feel free to move them around and we will review the changes; if something needs to change we can iterate 😄

Reply:
Ok, I take a try.

Reply:
By moving I meant moving all the files into a single package level `tree` not sub packages like `tree/avl`, `tree/binarysearchtree` etc.

Reply:
Got it. As per the above https://github.com/TheAlgorithms/Go/pull/551#discussion_r990012608, we should modify the `Node` to use generics. It takes time. I am willing to do it. There will be a one-day wait 😄

Reply:
No rush 😄

Reply:
Hi @tjgurwara99, I'm done 😀 . 
I have moved `avl`, `binarysearchtree` and `rbtree` to `tree` package. 
To avoid conflict, `binarysearchtree` has been wrapped into a struct: `BSTree`. 
As for the above discussed `Node`, inspired by https://github.com/emirpasic/gods/tree/master/trees , I didn't create an abstract version, but create three different versions: `Node` (for BSTree), `AVLNode` (for avl) and `RBNode` (for RBTree). 
For simplicity, I didn't change `avl`.